File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque
def solution(queue1, queue2):
    answer = -1
    n = len(queue1) + len(queue2)
    one = sum(queue1)
    two = sum(queue2)
    target = (one + two) / 2
    left = target - one
    right = target - two
    q1 = deque(queue1)
    q2 = deque(queue2)
    cnt = 0
    while q1 and q2:
        if left == 0 and right == 0:
            answer = cnt
            break
        if cnt > 2 * n:
            break
        if left > right:
            out = q2.popleft()
            left -= out
            right += out
            q1.append(out)
            cnt += 1
        elif left < right:
            out = q1.popleft()
            left += out
            right -= out
            q2.append(out)
            cnt += 1
            
    
    return answer

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;
class Solution {
    public long solution(int[] queue1, int[] queue2) {
        long answer = 0;
        Queue<Integer> q1 = new ArrayDeque<>();
        Queue<Integer> q2 = new ArrayDeque<>();
        long sum1 = 0, sum2 = 0;
        for(int i=0; i<queue1.length; i++){
            q1.add(queue1[i]);
            q2.add(queue2[i]);
            sum1 += queue1[i];
            sum2 += queue2[i];
        }
        while(sum1 != sum2){
            if(answer > 3 * queue1.length){
                answer = -1;
                break;
            }
            if(sum1 > sum2){
                int out = q1.remove();
                sum1 -= out;
                sum2 += out;
                q2.add(out);
                answer++;
            } else{
                int out = q2.remove();
                sum2 -= out;
                sum1 += out;
                q1.add(out);
                answer++;
            }
        }
        return answer;
    }
}